Tests/MongoSwiftTests/EventLoopBoundMongoClientTests.swift

```diff
@@ -8,53 +8,49 @@ final class EventLoopBoundMongoClientTests: MongoSwiftTestCase {
         let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
         let expectedEventLoop = elg.next()
 
-        try self.withTestClient { client in
+        try self.withTestClient(eventLoopGroup: elg) { client in
+            // TODO: SWIFT-1030 - add tests for database operations that return ChangeStream and MongoCursor
             let elBoundClient = EventLoopBoundMongoClient(client: client, eventLoop: expectedEventLoop)
-            expect(elBoundClient.db(Self.testDatabase).eventLoop) === expectedEventLoop
-        }
-    }
-
-    func testCollection() throws {
-        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
-        let expectedEventLoop = elg.next()
-
-        try self.withTestClient { client in
-            let elBoundClient = EventLoopBoundMongoClient(client: client, eventLoop: expectedEventLoop)
-
             let db = elBoundClient.db(Self.testDatabase)
-            defer { try? db.drop().wait() }
+            expect(db.eventLoop) === expectedEventLoop
+
+            // test the MongoDatabase operations return futures on the expected event loop
+            expect(db.createCollection("test").eventLoop) === expectedEventLoop
```

Comment:
can we ensure the collection this future evaluates to has the proper event loop too?

Reply:
Oh yeah, I also just added checks for the collections returned by `listMongoCollections`, but I wonder if that's redundant?

Reply:
I forgot to run `make linuxmain`

Reply:
No I don't think it's redundant, its a good test case :+1: